### PR TITLE
Collapsible scripts Addon

### LINKED ISSUE #1199: Collapsible blocks
**Describe the new addon**
Allows you to collapse hat, c, and reporter blocks.

**Link to the new addon**
I don't have one.

### PR DESCRIPTION
**Pull request by ErrorGamer2000**
**Resolves**

Resolves #1199

**Changes**

Added collapsible scripts addon, including saving when sprites are switched.

**Tests**

https://user-images.githubusercontent.com/70162741/116759868-a61ad480-a9d0-11eb-9412-4428a5f1c9ec.mp4

### CODE REVIEW

**Comment by TheColaber** [comment]
Scratch already has a context menu thing setup for this. Tbh all you need to do is just do `Blockly.getMainWorkspace().options.colapse = true` and then do `Blockly.Msgs.COLLAPSE_THING or (whatever it is) = "MSG"`

**Comment by TheColaber** [comment]
However, you have done this to individual blocks, which is not a scratch dead feature.

**Comment by TheColaber** [comment]
You also seem to have 2 addons in this PR which is not right.

**Comment by ErrorGamer2000** [comment]
Oh, oops. Sorry. I will fix it tomorrow.

**Comment by ErrorGamer2000** [comment]
Actually, I did it now.

**Comment by RedGuy12** [comment]
`createBlockContextMenu` won't be available until #1754 is merged

**Comment by ErrorGamer2000** [comment]
I know, but will you check my code to see if I got it right?

File: addons/script-collapse/addon.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "Collapsible scripts",
+  "description": "Enables script collapsing in the editor. ",
+  "tags": ["editor", "beta"],
+  "info": [
+    {
+      "type": "notice",
+      "text": "Other users will not be able to see collapsed scripts.",
```
Comment (RedGuy12) [inline addons/script-collapse/addon.json:8]: ofc? remove the notice

Comment (CST1229) [reply addons/script-collapse/addon.json:8]: Maybe the addon could add a comment like "collapsed" to remember collapsed scripts.

Comment (ErrorGamer2000) [reply addons/script-collapse/addon.json:8]: What do you mean?

Comment (ErrorGamer2000) [reply addons/script-collapse/addon.json:8]: > Maybe it's actually worth noting that if you reload the project, all the scripts will be uncollapsed, even if they were collapsed before.

Yeah, that is probably a good idea...even if you just hit `see project page`. I am thinking about using localstorage?

Comment (CST1229) [reply addons/script-collapse/addon.json:8]: What do you mean?

Comment (ErrorGamer2000) [reply addons/script-collapse/addon.json:8]: 🙄 That would require a TON of work.

Comment (CST1229) [reply addons/script-collapse/addon.json:8]: Yeah.

**Comment by TheColaber** [comment]
Was there a reason this was closed?

**Comment by RedGuy12** [comment]
because it's blocked by #1754

**Comment by TheColaber** [comment]
So? That PR is still open.

**Comment by RedGuy12** [comment]
#1754 was merged.

@ErrorGamer2000 if you want you can make a new pr

**Pull request closed without merging**